boot/bootutil/src/loader.c

```diff
@@ -318,8 +318,8 @@ boot_write_sz(void)
      * on what the minimum write size is for scratch area, active image slot.
      * We need to use the bigger of those 2 values.
      */
-    elem_sz = hal_flash_align(boot_img_fa_device_id(&boot_data, 0));
-    align = hal_flash_align(boot_scratch_fa_device_id(&boot_data));
```

Comment:
`boot_scratch_fa_device_id` is not used anymore, please remove the definition. Probably ok to remove `boot_img_fa_device_id` as well, and expand the only use of it (on `loader.c`).